core/src/crds_value.rs

```diff
@@ -12,6 +13,58 @@ pub enum CrdsValue {
     ContactInfo(ContactInfo),
     /// * Merge Strategy - Latest wallclock is picked
     Vote(Vote),
+    /// * Merge Strategy - Latest wallclock is picked
+    EpochSlots(EpochSlots),
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub struct EpochSlots {
+    pub from: Pubkey,
+    pub root: u64,
+    pub slots: HashSet<u64>,
```

Comment:
is this ugly?  how is a HashSet serialized?

Reply:
It should be efficient, the Serialize trait is implemented for HashSet, and should map pretty closely to a vector according to docs: https://serde.rs/data-model.html